src/main/java/com/productdock/library/inventory/book/BookService.java

```diff
@@ -0,0 +1,29 @@
+package com.productdock.library.inventory.book;
+
+import com.productdock.library.inventory.domain.Book;
+import com.productdock.library.inventory.producer.Publisher;
+import com.productdock.library.inventory.record.RentalRecord;
+import org.springframework.stereotype.Service;
+
+
+@Service
+public record BookService(BookRepository bookRepository, Publisher publisher, BookMapper bookMapper) {
+
+    public void updateBookState(RentalRecord rentalRecord) throws Exception{
+        saveNewBook(rentalRecord);
+        var optionalBook = bookRepository.findById(rentalRecord.getBookId());
+        if (optionalBook.isPresent()) {
+            var bookEntity = optionalBook.get();
+            Book book = bookMapper.toDomain(bookEntity);
+            book.processRentalRecords(rentalRecord);
+            bookRepository.save(bookMapper.toEntity(book));
+            publisher.sendMessage(BookAvailabilityMessage.builder().bookId(book.getBookId()).availableBookCount(book.getAvailableBooksCount()).build());
+        }
+    }
+
+    private void saveNewBook(RentalRecord rentalRecord) {
```

Comment:
This is unexpected behavior, update should update existing entity. If there is no entity it should throw an exception. Especially because we can’t guess number of book copies library has (ok, we can guess 1 and it will most likely be true but it is not ok from design perspective).
Double check how Kafka will behave if message processing fails, will it retry it?